x-pack/plugins/maps/public/actions/data_request_actions.ts

```diff
@@ -45,22 +45,28 @@ import {
 } from './map_action_constants';
 import { ILayer } from '../classes/layers/layer';
 import { IVectorLayer } from '../classes/layers/vector_layer';
-import { DataMeta, MapExtent, MapFilters } from '../../common/descriptor_types';
+import { DataRequestMeta, MapExtent, DataFilters } from '../../common/descriptor_types';
 import { DataRequestAbortError } from '../classes/util/data_request';
 import { scaleBounds, turfBboxToBounds } from '../../common/elasticsearch_util';
 
 const FIT_TO_BOUNDS_SCALE_FACTOR = 0.1;
 
 export type DataRequestContext = {
-  startLoading(dataId: string, requestToken: symbol, requestMeta?: DataMeta): void;
-  stopLoading(dataId: string, requestToken: symbol, data: object, resultsMeta?: DataMeta): void;
+  startLoading(dataId: string, requestToken: symbol, requestMeta?: DataRequestMeta): void;
+  stopLoading(
+    dataId: string,
+    requestToken: symbol,
+    data: object,
+    resultsMeta?: DataRequestMeta
+  ): void;
   onLoadError(dataId: string, requestToken: symbol, errorMessage: string): void;
   onJoinError(errorMessage: string): void;
   updateSourceData(newData: unknown): void;
   isRequestStillActive(dataId: string, requestToken: symbol): boolean;
   registerCancelCallback(requestToken: symbol, callback: () => void): void;
-  dataFilters: MapFilters;
-  forceRefresh: boolean;
+  dataFilters: DataFilters;
+  forceRefreshDueToDrawing: boolean;
```

Comment:
Do we need flags for both forceRefreshDueToDrawing and isForceRefresh? How about just a single flag, isForceRefresh?

Reply:
Thanks for explaining the different uses of the flags, It makes sense to keep them both. Can you please update these flags with comments on why they both are needed so we do not lose this information

// Boolean signaling data request triggered by a user updating layer features via drawing tools. When true, layer will re-load regardless of source applyForceRefresh flag.
forceRefreshDueToDrawing: boolean;

// Boolean signaling data request triggered by auto-refresh timer or user clicking refresh button. When true, layer will re-load only when source applyForceRefresh flag is set to true.
isForceRefresh